skimage/transform/integral.py

```diff
@@ -21,15 +21,28 @@ def integral_image(image):
     S : ndarray
         Integral image/summed area table of same shape as input image.
 
+    Notes
+    -----
+    For better accuracy and to avoid potential overflow, the datatype of the
+    output may differ from the input when the default dtype of None is used.
+    For inputs with integer dtype, the behavior matches that for
+    :func:`numpy.cumsum`. Floating point inputs will be promoted to at least
+    double precision. The user can set `dtype` to override this behavior.
+
     References
     ----------
     .. [1] F.C. Crow, "Summed-area tables for texture mapping,"
            ACM SIGGRAPH Computer Graphics, vol. 18, 1984, pp. 207-212.
 
     """
+    if dtype is None and image.real.dtype.kind == 'f':
+        # default to at least double precision cumsum for accuracy
+        dtype = np.promote_types(image.dtype, float)
```

Comment:
I think on some Windows builds, "float" will default to float32. Should we specify np.float64 here?

Reply:
That is the case for `int` on 32-bit, but I thought float was usually double precision? Definitely not a problem to just use np.float64 to make it unambiguous, though